docs/lang/articles/basic/type.md

```diff
@@ -1,217 +1,188 @@
 ---
 sidebar_position: 3
+
 ---
 
 # Type system
 
-Data types in Taichi consist of _primitive types_ and _compound types_. Primitive types are the numerical data types used by different backends, while compound types are user-defined types of data records composed of multiple members.
+Taichi supports two data types in the [Taichi scope](./syntax.md#taichi-scope): primitive type and compound type.
+
+- Primitive types: Various commonly-used numerical data types, such as `int32`, `uint8`, and `float64`
+- Compound types: User-defined data types, which comprise multiple members.
 
 ## Primitive types
 
-Taichi supports common numerical data types as its primitive types. Each type is denoted as a
-character indicating its _category_ followed by a number indicating its _precision bits_. The
-_category_ can be either `i` (for signed integers), `u` (for unsigned integers), or `f` (for floating-point numbers). The _precision bits_ can be either `8`, `16`, `32`, or `64`,
-which represents the number of **bits** for storing the data. For example, the two most commonly used types:
+Each primitive type is denoted with a character indicating its category followed by a number indicating its precision bits (number of bits for storing the data). The _category_ can be `i` (signed integers), `u` (unsigned integers), or `f` (floating-point numbers); the _precision bits_ can be `8`, `16`, `32`, or `64`. Following are the two most commonly-used types:
+
+- `i32`: 32-bit signed integer
+- `f32` : 32-bit floating-point number.
+
+### Supported primitive types
+
+Not all primitive types are supported by your backend. Check out the following table for the supported types. Note that some backends may require extensions to support a specific primitive type.
 
-- `i32` represents a 32-bit signed integer;
-- `f32` represents a 32-bit floating-point number.
+| Backend | `i8`               | `i16`              | `i32`              | `i64`              | `u8`                 | `u16`                | `u32`                | `u64`                | `f16`                | `f32`                | `f64`                |
+| ------- | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ |
+| CPU     | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
```

Comment:
nits:

* heavy_check_mark is dark gray, better use a green color
* `x` and `o` are of the same color, better distinguish the two

Reply:
Searched for available markdown emojis but failed to find better options. Have asked the frontend developer to come up with CSS for these two emojis.